### ISSUE TITLE
thread 'main' panicked at 'app config table has missing or extra value

### ISSUE THREAD

**Issue by rubo77** [freifunk-saar/ff-node-monitor#7]
```
# RUST_BACKTRACE=1 /var/lib/ff-node-monitor/src/target/release/ff-node-monitor
🔧
    => address: 127.0.0.1
    => port: 8833
    => log: normal
    => workers: 2
    => secret key: provided
    => limits: forms = 32KiB
    => tls: disabled
    => [extra] ff-node-monitor: { secrets = { action_signing_key = "4ba6997a1469927798f1461fa06d0062ef2224c1bbe3f4c265112e151f8dab09", postgres_url = "postgres://ff-node-monitor@/ff-node-monitor" }, ui = { email_from = "freifunk@example.de", instance_name = "Freifunk Kiel" }, urls = { nodes_url = "https://map.freifunk.in-kiel.de/yanic/nodes.json", root_url = "https://localhost" } }
thread 'main' panicked at 'app config table has missing or extra value: Error { inner: ErrorInner { kind: Custom, line: None, col: 0, message: "missing field `root`", key: ["urls"] } }', libcore/result.rs:945:5
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:511
   5: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:426
   6: rust_begin_unwind
             at libstd/panicking.rs:337
   7: core::panicking::panic_fmt
             at libcore/panicking.rs:92
   8: core::result::unwrap_failed
   9: ff_node_monitor::config::Config::new
  10: ff_node_monitor::config::fairing::{{closure}}
  11: <rocket::fairing::ad_hoc::AdHoc as rocket::fairing::Fairing>::on_attach
  12: rocket::fairing::fairings::Fairings::attach
  13: rocket::rocket::Rocket::attach
  14: ff_node_monitor::main
  15: std::rt::lang_start::{{closure}}
  16: std::panicking::try::do_call
             at libstd/rt.rs:59
             at libstd/panicking.rs:310
  17: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:105
  18: std::rt::lang_start_internal
             at libstd/panicking.rs:289
             at libstd/panic.rs:374
             at libstd/rt.rs:58
  19: main
  20: __libc_start_main
  21: <unknown>
```

I use this Rocket.toml
```
[global.ff-node-monitor.ui]
# The name of your Freifunk community.
instance_name = "Freifunk Kiel"
# The sender address of the emails that are sent by ff-node-monitor.  Make sure
# you have a mail service running locally that is set up to deliver emails from
# that address.
email_from = "freifunk@example.de"

[global.ff-node-monitor.urls]
# The root URL where you will be hosting ff-node-monitor.
root_url = "https://localhost"
# The URL to the hopglass nodes.json file.
nodes_url = "https://map.freifunk.in-kiel.de/yanic/nodes.json"

[global.ff-node-monitor.secrets]
# PostgreSQL credentials.  If you followed the instructions in the README, this
# should be correct.
postgres_url = "postgres://ff-node-monitor@/ff-node-monitor"
# Key used to sign data for confirmation emails.  Generate this key with
# `openssl rand -hex 32`.
action_signing_key = "REDACTED"

[global]
# The address and port on which ff-node-monitor will listen.
address = "127.0.0.1"
port = 8833
# Secret key used by Rocket.  Generate this key with `openssl rand -base64 32`.
secret_key = "REDACTED"
```

**Comment by RalfJung** [comment]
I removed the secret keys from your configuration, but it may be a good idea for you to also change them on your side before you go live.

**Comment by RalfJung** [comment]
Ah, I forgot to update the `Rocket.toml.dist` when I renamed some config keys.

The two items in the `[urls]` section should be called `root` and `nodes`. Fixed by a2f7db055fb774171d9c18d2f7264bc888c45307.

Sorry that you are running into all of these issues, I guess I should have tried installing this on a cleam system before declaring the instructions done. Thanks for your patience in reporting them!
